processor/resourcedetectionprocessor/README.md

```diff
@@ -17,6 +17,14 @@ details of which are currently pending confirmation in the OpenTelemetry specifi
     * host.name
     * os.type
 
+system detector custom configuration example:
+```yaml
+detectors: ["system"]
+system:
+    # Force the detector to use 'hostname' instead of FQDN
+    use_hostname: true
```

Comment:
What about something like this?
```suggestion
    system_source: hostname
```

That way if we add further options to the system detector the new option will still be useful. And we can have an `auto` default for the current behavior (try FQDN and fall back to hostname). Does that make sense to you? (I am not super convinced myself so please push back if this has some flaw 😄)

Reply:
Yes, that makes sense. I have added the suggested changes.

Reply:
I don't think that `system_source` is a proper name. It's a source for fetching `host.name` attribute value. 

Can we maybe call it `hostname_sources` and have the value as a list of sources in order of fetching priority. The default will be ["fqdn", "hostname"]. For the use case when FQDN is not needed we set it to `hostname_sources: ["hostname"]`. 

@mx-psi @hprajapati-splunk what do you think?

Reply:
We can consider other names for the sources values like `hostname_sources: ["dns", "os"]` for the default. Not sure tho, happy to discuss other options

Reply:
`hostname_sources` sounds better to me than `system_source`and passing a priority list makes sense, they both seem like good suggestions

Reply:
ok, sounds good. if everyone agrees, @hprajapati-splunk let's use the config with `hostname_sources: ["dns", "os"]` by default

Reply:
@dmitryax @mx-psi 
If we use a priority list then few questions arise like:
    1. What if any value in the given list is invalid like: `["dns", "invalid"]` ?
    2. If all the values in the priority list failed for ex. `["dns", "os"]` is the passed list but both methods failed to fetch the 
        required value then shall we return one error message or all the errors that we got from all the methods(errors from 
        methods which fetch `FQDN` and `hostname`?

Reply:
>  1. What if any value in the given list is invalid like: ["dns", "invalid"] ?

I think the most sensible way is to check this at configuration validation time and return an error if there are any invalid entries.

> 2. If all the values in the priority list failed for ex. ["dns", "os"] is the passed list but both methods failed to fetch the required value then shall we return one error message or all the errors that we got from all the methods(errors from methods which fetch FQDN and hostname?

Yes, you can use `consumererror.Combine` for this.

Reply:
@dmitryax @mx-psi 
I have added the suggested changes. Please, review the same.